### DISCUSSION TITLE
SavvyCAN for regression testing

### DISCUSSION THREAD

**Discussion by mickeyl** [collin80/SavvyCAN#322] [Q&A]
I wonder whether I can use SavvyCAN for regression testing of my reprogramming software.

Ideally I would be able to capture a complete UDS communication sequence and then
SavvyCAN would stand in as one of the participants, i.e. listen for the request and reply
with the appropriate responses (as captured).

If the request does not match the expected "next" request, SavvyCAN would just respond
with a generic error.

Is that possible right now (perhaps without resorting to scripting?) or, if not, something you would think is worthwhile?

**Comment by collin80** [comment]
It isn't possible right at the moment without significant scripting. With scripting I suppose it could be done but essentially you'd have to manually create scripts that implement one side of the communication. It is an interesting idea to allow for being one side of the connection. If you're able to use Windows you could investigate BusMaster which does support generating code to be an ECU simulator. I believe this is via C++ plugins though so not any easier. 

I can see many uses for this, not only regression testing. It'd also be really useful for people who are taking parts out of a car and trying to run them on the bench. It'd be very nice to have SavvyCAN fake being the missing nodes on the bus.
